src/constructs/aws/Queue.ts

```diff
@@ -206,17 +206,31 @@ export class Queue extends AwsConstruct {
             ...encryption,
         });
 
-        const alarmEmail = configuration.alarm;
-        if (alarmEmail !== undefined) {
-            const alarmTopic = new Topic(this, "AlarmTopic", {
-                topicName: `${this.provider.stackName}-${id}-dlq-alarm-topic`,
-                displayName: `[Alert][${id}] There are failed jobs in the dead letter queue.`,
-            });
-            new Subscription(this, "AlarmTopicSubscription", {
-                topic: alarmTopic,
-                protocol: SubscriptionProtocol.EMAIL,
-                endpoint: alarmEmail,
-            });
+        if (configuration.alarm !== undefined) {
+            let alarmArn: string;
+            let alarmExistingTopic = undefined;
+            if (configuration.alarm.startsWith("arn:aws:sns:")) {
```

Comment:
This would resolve only if user specify an hardcoded string as alarm topic. Any serverless resolvers used in place of an hardcoded string will fail this check.

Reply:
I just mocked up the following and it seemed to work fine per the cloudformation output from eject and did not throw a serverless error. But I may not understand what "serverless resolvers" means.

serverless.yml
```
custom:
  alarmSnsArn: arn:aws:sns:${aws:region}:${aws:accountId}:alarm-alerts

constructs:
  send:
    type: queue
    alarm: ${self:custom.alarmSnsArn}
```

cloudformation eject
```
  sendAlarmDFF3285F:
    Type: 'AWS::CloudWatch::Alarm'
    Properties:
      ComparisonOperator: GreaterThanThreshold
      EvaluationPeriods: 1
      AlarmActions:
        - 'arn:aws:sns:us-east-1:123456789012:alarm-alerts'
```

Is an example of a serverless resolver something like pulling in a CF output from another stack `alarm: !ImportValue another-stack-${self:provider.stage}-OutputArn`? this does throw a messy error currently it seems.

Is there anywhere else in lift this works I could use as an example to fix here?

Reply:
It wouldn't work with, for example, `!Ref SomeTopic`